src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/ZipVerifiers.java

```diff
@@ -148,6 +186,26 @@ public ZipStreamWithSignature(
             this.zipFileName = zipFileName;
             this.container = container;
         }
+
+        public static ZipStreamWithSignature fromKeyfile(
+            ZipInputStream zipInputStream,
+            String publicKeyDerFile,
+            String zipFileName,
+            String container
+        ) {
+            String publicKeyBase64 = null;
+            try {
+                if (!Strings.isNullOrEmpty(publicKeyDerFile) && !"none".equalsIgnoreCase(publicKeyDerFile)) {
+                    publicKeyBase64 =
+                        Base64.getEncoder().encodeToString(
+                            Resources.toByteArray(getResource(publicKeyDerFile))
+                        );
```

Comment:
It would be nice to read this file only once, on application start.
Happy to do have it done on a separate PR.